src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -231,8 +242,19 @@ int halide_hexagon_remote_run(handle_t module_ptr, handle_t function,
         *next_arg = input_scalarsPtrs[i].data;
     }
 
+    // Prior to running the pipeline, power HVX on (if it was not already on).
+    int result = halide_hexagon_remote_power_hvx_on();
+    if (result != 0) {
+        return result;
+    }
+
     // Call the pipeline and return the result.
-    return run_context.run(pipeline, args);
+    result = run_context.run(pipeline, args);
+
+    // Power HVX off.
+    halide_hexagon_remote_power_hvx_off();
```

Comment:
The idea is that power_hvx_on/off calls can be called nested with each other, and only the outermost calls will actually do anything. The calls here are basically a last resort to make sure the program works at all (and for long running kernels, are probably sufficient). Programs can optionally call power_hvx_on/off themselves at a higher level in the program to eliminate the overhead.